Add Tests for `get_titles` function

Animesh-Ghosh opened the pull request:
## Description

This PR introduces a test suite for the project along with some minor documentation and script updates.
Was not able to figure out how exactly to go about testing the `update_titles` function, so would appreciate feedback there.

## Changes Include

1. Updated `main.py` script to have a docstring and only run the `update_titles` function if the script was run directly.
2. Added initial tests for the `get_titles` function in `test_titles_updater.py`.

<!-- Please delete options that are not relevant  -->

- [x] Refactor ⚒️ (code change that neither fixes a bug nor adds a feature)

## How Has This Been Tested?

## Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Screenshots (if any)

Animesh-Ghosh committed:
Add Tests for get_titles function

Added docstring for main.py script and updated it to only invoke update_titles
function if the file was run directly.
Added initial tests for titles updater script.

Animesh-Ghosh committed:
Update note for future contributors

Updated note for contributors/reviewers.

Bhupesh-V commented on test_titles_updater.py:
```diff
@@ -0,0 +1,29 @@
+
+from main import get_titles
+
+# NOTE: not sure about how exactly update_titles can be tested
+class TestTitlesUpdater(unittest.TestCase):
```
```suggestion
class TestGetTitles(unittest.TestCase):
```

Bhupesh-V reviewed (changes requested):
Hey thanks a lot for adding tests @Animesh-Ghosh 

Can you add steps to execute this tests in the README.md?

Also for mocking API resuests [checkout this thread](https://www.reddit.com/r/redditdev/comments/q2d5bc/unit_tests_for_praw/). I am also not sure ATM on how to achieve this. If it takes too much effort from your side, we can let this go for the moment

Animesh-Ghosh committed:
Update README with instructions to run tests

Updated README with instructions to run the tests.
Renamed test file and class.

Bhupesh-V commented:
@all-contributors please add @Animesh-Ghosh  for code, test
